applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/explicit_strategy.hpp

```diff
@@ -0,0 +1,537 @@
+    {
+        mCalculateReactionsFlag = CalculateReactionsFlag;
+    }
+
+    bool GetCalculateReactionsFlag()
+    {
+        return mCalculateReactionsFlag;
+    }
+
+    void SetReformDofSetAtEachStepFlag(bool flag)
+    {
+
+          mReformDofSetAtEachStep = flag;
+
+    }
+
+    bool GetReformDofSetAtEachStepFlag()
+    {
+        return mReformDofSetAtEachStep;
+    }
+
+    //level of echo for the solving strategy
+    // 0 -> mute... no echo at all
+    // 1 -> printing time and basic informations
+    // 2 -> printing linear solver data
+    // 3 -> Print of debug informations:
+    //    Echo of stiffness matrix, Dx, b...
+
+    void SetEchoLevel(int Level)
+    {
+        BaseType::mEchoLevel = Level;
+
+    }
+
+    //**********************************************************************
+    //**********************************************************************
+
+    void Initialize()
+    {
+        KRATOS_TRY
+        //pointers needed in the solution
+        typename TSchemeType::Pointer pScheme                     = GetScheme();
+        typename TBuilderAndSolverType::Pointer pBuilderAndSolver = GetBuilderAndSolver();
+        ModelPart& r_model_part                                   = BaseType::GetModelPart();
+
+        TSystemMatrixType mA = TSystemMatrixType();   //dummy initialization. Not used in builder and solver.
+
+        //Initialize The Scheme - OPERATIONS TO BE DONE ONCE
+        if (pScheme->SchemeIsInitialized() == false)
+            pScheme->Initialize(BaseType::GetModelPart());
+        
+        //Initialize The Elements - OPERATIONS TO BE DONE ONCE
+        if (pScheme->ElementsAreInitialized() == false)
+            pScheme->InitializeElements(BaseType::GetModelPart());
+
+        //Initialize The Conditions- OPERATIONS TO BE DONE ONCE
+        if (pScheme->ConditionsAreInitialized() == false)
+            pScheme->InitializeConditions(BaseType::GetModelPart());
+
+        pBuilderAndSolver->BuildLHS(pScheme, r_model_part, mA); //calculate Mass Matrix
+        TSystemVectorType mb  = TSystemVectorType();
+        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), mb); //calculate Residual for scheme initialization
+        
+        mInitializeWasPerformed = true;
+
+	    //std::cout<<" Rebuild Level "<<BaseType::mRebuildLevel<<std::endl;
+
+        KRATOS_CATCH( "" )
+    }
+
+    //**********************************************************************
+    //**********************************************************************
+    
+    void InitializeSolutionStep()
+    {
+        KRATOS_TRY
+        typename TSchemeType::Pointer pScheme                     = GetScheme();
+        typename TBuilderAndSolverType::Pointer pBuilderAndSolver = GetBuilderAndSolver();
+        ModelPart& r_model_part                                   = BaseType::GetModelPart();
+
+        TSystemMatrixType mA = TSystemMatrixType();   //dummy initialization. Not used in builder and solver.
```

Comment:
please use the name dummy in the name of the variable.
Tracing it becomes difficult otherwise